extensions/resteasy-reactive/rest-client-reactive/runtime/src/main/java/io/quarkus/rest/client/reactive/runtime/MicroProfileRestClientResponseFilter.java

```diff
@@ -29,21 +33,53 @@ public MicroProfileRestClientResponseFilter(List<ResponseExceptionMapper<?>> exc
     public void filter(ClientRequestContext requestContext, ClientResponseContext responseContext) throws IOException {
         for (ResponseExceptionMapper exceptionMapper : exceptionMappers) {
             if (exceptionMapper.handles(responseContext.getStatus(), responseContext.getHeaders())) {
-                // we have an exception mapper, we don't need the response anymore, we can map it to response right away (I hope :D)
                 RestClientRequestContext restClientContext = ((ClientRequestContextImpl) requestContext)
                         .getRestClientRequestContext();
-                ResponseImpl response = ClientResponseCompleteRestHandler.mapToResponse(restClientContext, false);
-                Throwable throwable;
-                if (exceptionMapper instanceof ResteasyReactiveResponseExceptionMapper) {
-                    throwable = ((ResteasyReactiveResponseExceptionMapper) exceptionMapper).toThrowable(response,
-                            restClientContext);
+
+                boolean requiresBlocking = RestClientRecorder.isClassBlocking(exceptionMapper.getClass());
+                if (Context.isOnEventLoopThread() && requiresBlocking) {
+                    switchToWorkerThreadPoolAndRetry(restClientContext);
+                    break;
                 } else {
-                    throwable = exceptionMapper.toThrowable(response);
-                }
-                if (throwable != null) {
-                    throw new UnwrappableException(throwable);
+                    // we have an exception mapper, we don't need the response anymore, we can map it to response right away (I hope :D)
+                    ResponseImpl response = ClientResponseCompleteRestHandler.mapToResponse(restClientContext, false);
+                    Throwable throwable;
+                    if (exceptionMapper instanceof ResteasyReactiveResponseExceptionMapper) {
+                        throwable = ((ResteasyReactiveResponseExceptionMapper) exceptionMapper).toThrowable(response,
+                                restClientContext);
+                    } else {
+                        throwable = exceptionMapper.toThrowable(response);
+                    }
+                    if (throwable != null) {
+                        throw new UnwrappableException(throwable);
+                    }
                 }
             }
         }
     }
+
+    private void switchToWorkerThreadPoolAndRetry(RestClientRequestContext restClientContext) {
+        List<ClientRestHandler> nextHandlers = new ArrayList<>();
```

Comment:
Can we specify the capacity of the lists being used in this method?
We already know what we will be added to them, so there is no reason to cause excess memory allocation :)

It obviously won't have any real effect, but we do try to do this wherever possible as a very large bunch of tiny such cases can add up

Reply:
Done